### PR TITLE
Fix bug in handling of invalid if-modified-since/last-modified values

### PR DESCRIPTION
**Pull request by benurb**
Hi @dougwilson 

thanks for this very useful lib, but unfortunately I found two little bugs.
1. The tests for invalid values in if-modified-since/last-modified were broken and therefore covering up ...
2. a actual bug in the handling of invalid values in if-modified-since headers.
I split up the changes in two commits so it's easier for you the reproduce the behavior. With only the test changes applied you can see that one test is actually failing and with both commits applied all tests should be green again 👍 

Best regards,
Ben

### CODE REVIEW

**Comment by benurb** [comment]
Oh, ok. Didn't know about the `Date.parse` problem. What solution do you prefer?
A long if clause?
```js
var lastModifiedTimestamp = Date.parse(resHeaders['last-modified'])
var modifiedSinceTimestamp = Date.parse(modifiedSince)
var modifiedStale = !lastModifiedTimestamp || isNaN(lastModifiedTimestamp) ||
  !modifiedSinceTimestamp || isNaN(modifiedSinceTimestamp) ||
  !lastModifiedTimestamp > modifiedSinceTimestamp
```
Or a short circuit?
```js
var lastModifiedTimestamp = Date.parse(resHeaders['last-modified']) || NaN
var modifiedSinceTimestamp = Date.parse(modifiedSince) || NaN
var modifiedStale = isNaN(lastModifiedTimestamp) || isNaN(modifiedSinceTimestamp) ||
  lastModifiedTimestamp > modifiedSinceTimestamp
```

**Comment by dougwilson** [comment]
Neither of them seem to work correctly for the one date / time in which Date.parse returns 0 (unix epoch).

**Comment by benurb** [comment]
Hey @dougwilson,
yep, you're right. I've change the code a bit and added tests for all the edge cases you mentioned.

File: test/fresh.js [hunk 1]
```diff
@@ -107,18 +107,73 @@ describe('fresh(reqHeaders, resHeaders)', function () {
...
+    describe('with invalid Last-Modified date', function () {
       it('should be stale', function () {
         var reqHeaders = { 'if-modified-since': 'Sat, 01 Jan 2000 00:00:00 GMT' }
-        var resHeaders = { 'modified-since': 'foo' }
+        var resHeaders = { 'last-modified': 'foo' }
+        assert.ok(!fresh(reqHeaders, resHeaders))
+      })
+    })
+
+    describe('with earliest possible If-Modified-Since date', function () {
```
Comment (dougwilson) [inline test/fresh.js]: I'm not sure if this makes sense, because you can use earlier dates / times, so not sure why this is the "earliest possible".

File: test/fresh.js [hunk 2]
```diff
@@ -107,18 +107,73 @@ describe('fresh(reqHeaders, resHeaders)', function () {
...
+
+    describe('with earliest possible Last-Modified and If-Modified-Since date', function () {
+      it('should be fresh', function () {
+        var reqHeaders = { 'if-modified-since': 'Thu, 01 Jan 1970 00:00:00 GMT' }
+        var resHeaders = { 'last-modified': 'Thu, 01 Jan 1970 00:00:00 GMT' }
+        assert.ok(fresh(reqHeaders, resHeaders))
+      })
+    })
+
+    describe('when Date.parse was monkey patched', function () {
```
Comment (dougwilson) [inline test/fresh.js]: If this is going to actually have a test, just test with what we really care about specifically: the datejs patching.

Comment (benurb) [inline test/fresh.js]: Mhh ok. I haven't have a look at datejs yet, but do you suggest adding it as a devDependency or rather make the mock more similar to the datejs patching? Or are these tests a bad idea in general?

File: test/fresh.js (hunk 1 above)
Comment (benurb) [inline test/fresh.js]: Yeah you're right. I'll change that to something like 'with unix epoch in If-Modified-Since', ok?

File: test/fresh.js (hunk 2 above)
Comment (dougwilson) [inline test/fresh.js]: A devDep is just fine, as they are available during the testing. If you want to just not have any tests, that's OK too, but if you wanted to make some patching tests, I'd say to either (a) directly use datejs in them or (b) make the mocking patches actually function like the datejs does (which afaik these are different from what datejs returns).

Comment (benurb) [inline test/fresh.js]: I was not able use datejs without polluting global objects, but I updated to mock to return the same result for the strings in the tests. If you don't like this I would rather remove the tests again.

**Comment by benurb** [comment]
Hi @dougwilson 
I addressed your comments. Let me know what you think 👍

**Comment by benurb** [comment]
Hey @dougwilson 
any additional thoughs on this PR?

**Comment by dougwilson** [comment]
Hi @benurb I'm really sorry I let this fall though, truly sorry. I just got it merged in now.
